Notifier-service/src/interfaces/models/conference.ts

```diff
@@ -15,7 +15,7 @@ import { Model } from '../model';
 @injectable()
 export abstract class ConferenceModel extends Model{
     protected abstract modelName:string;
-    abstract async getOne():Promise<ConferenceDocument | null>
+    abstract async getOne(query:any):Promise<ConferenceDocument | null>
```

Comment:
Here u are just mentioning the query and not both queryKey and queryValue as mentioned https://github.com/rajatkb/Conference-Notify/pull/91/files#diff-a37f49d21bf5d0b42197cde809059e30R13. Are u using only value or both key and value as parameter?

Reply:
Query is an object of key value pairs. This method is implemented in class ConferenceModelMongo to query mongoDB based on the query passed, by default the value is an empty object.
I have further called this function in getOne function of ConferenceServiceI class. In this function, I take queryKey and queryValue as input, create the query object and call the getOne function of ConferenceModelMongo object

Reply:
Hence both queryKey and queryValue are used

Reply:
* Specify type to getOne argument. In our case, it will be a string for the _id. I got the argument you provided, but the way you are letting the data flow breaks the idea of localising the knowledge of database being used. Not that we will be hot-swapping different implementation, but current design allows to do that. Models getOne should clearly mention in its signature what it requires the services should not be aware of how to form a query.